framework/web/UrlManager.php

```diff
@@ -126,11 +126,14 @@ class UrlManager extends Component
      */
     public $ruleConfig = ['class' => 'yii\web\UrlRule'];
 
+    /**
+     * @var array internal cache to store matched rules
+     */
+    protected $ruleCache;
```

Comment:
Check out this article: http://fabien.potencier.org/pragmatism-over-theory-protected-vs-private.html
>Having a few well defined extension points force the developer to extend your library the right way instead of hacking your code.

In general, refactoring to allow access to previously private properties has been done many times in the past for Yii. If it makes sense and is not a very rare use case it will be implemented if requested.